compiler/rustc_abi/src/lib.rs

```diff
@@ -298,6 +323,7 @@ impl TargetDataLayout {
     /// determined from llvm string.
     pub fn parse_from_llvm_datalayout_string<'a>(
         input: &'a str,
+        default_address_space: AddressSpace,
```

Comment:
IMO this should be part of the data layout but currently none of alloca/globals/program AS can be used for this, so passing it in seems reasonable.